### PR TITLE
Modify test_respin_app_pod_exported_nfs_volume_incluster to address teardown of deployment pod and pvc

### LINKED ISSUE #9172: test_toleration fails due to pod nfs-test-pod-1-66lgv being stuck at terminating state
Message: assert False

where False = wait_for_pods_to_be_running(timeout=360)
Type: None

Text:
self = <tests.manage.z_cluster.nodes.test_toleration.TestTaintAndTolerations object at 0x7f5ac46dfcd0>

def test_toleration(self):
    """
    1. Check if nodes are tainted
    2. Taint ocs nodes if not tainted
    3. Check for tolerations on all pod
    4. Respin all ocs pods and check if it runs on ocs nodes
    5. Untaint nodes

    """
    # taint nodes if not already tainted
    nodes = get_ocs_nodes()
    for node in nodes:
        taint_nodes([node.name])

    # Check tolerations on pods under openshift-storage
    check_toleration_on_pods()

    # Respin all pods and check it if is still running
    pod_list = get_all_pods(namespace=config.ENV_DATA["cluster_namespace"])
    for pod in pod_list:
        if "s3cli" in pod.name:
            continue
        else:
            pod.delete(wait=False)

  assert wait_for_pods_to_be_running(timeout=360)
E       assert False
E        +  where False = wait_for_pods_to_be_running(timeout=360)

MG log:https://url.corp.redhat.com/d4bd367

### PR DESCRIPTION
**Pull request by pintojoy**
Updated test_respin_app_pod_exported_nfs_volume_incluster to handle deployment deletion and pvc deletion if test fails. Fixes #9172

### CODE REVIEW

**Review by keesturam — DISMISSED** [review]

**Comment by keesturam** [comment]
LGTM

File: ocs_ci/templates/app-pods/nfs_test_app.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+apiVersion: apps.openshift.io/v1
+kind: DeploymentConfig
+metadata:
+  name: nfs-test-pod
+  namespace: openshift-storage
```
Comment (prsurve) [inline ocs_ci/templates/app-pods/nfs_test_app.yaml:6]: we should not hardcode namespace

File: tests/functional/nfs_feature/test_nfs_feature_enable_for_ODF_clusters.py [hunk 2]
```diff
@@ -1132,174 +1132,137 @@ def test_respin_app_pod_exported_nfs_volume_incluster(
...
-            size="4G",
-            fio_filename=file_name,
-            runtime=60,
-        )
-        log.info("IO started on all pods")
+            file_name = pod_obj.name
+            # Run IO
+            pod_obj.run_io(
+                storage_type="fs",
+                size="4G",
```
Comment (prsurve) [inline tests/functional/nfs_feature/test_nfs_feature_enable_for_ODF_clusters.py:1173]: 4Gi

**Comment by pintojoy** [comment]
Test job passed here : https://url.corp.redhat.com/6959dc2

File: ocs_ci/templates/app-pods/nfs_test_app.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+---
+apiVersion: apps.openshift.io/v1
+kind: DeploymentConfig
```
[the remark this answers is not in the document]
Comment (pintojoy) [reply ocs_ci/templates/app-pods/nfs_test_app.yaml]: Made necessary changes

File: tests/functional/nfs_feature/test_nfs_feature_enable_for_ODF_clusters.py [hunk 4]
```diff
@@ -1132,174 +1132,137 @@ def test_respin_app_pod_exported_nfs_volume_incluster(
...
-            resource_count=1,
-            condition=constants.STATUS_RUNNING,
-            selector="name=nfs-test-pod",
-            dont_allow_other_resources=True,
-            timeout=60,
-        )
-        pod_objs = pod.get_all_pods(
-            namespace=self.namespace, selector=["nfs-test-pod"], selector_label="name"
-        )
+            # Create deployment config for app pod
```
Comment (ebenahar) [inline tests/functional/nfs_feature/test_nfs_feature_enable_for_ODF_clusters.py:1147]: in OCP 4.14 DeploymentConfigs were deprecated - https://docs.openshift.com/container-platform/4.15/applications/deployments/what-deployments-are.html
Please use Deployments instead

Comment (pintojoy) [reply tests/functional/nfs_feature/test_nfs_feature_enable_for_ODF_clusters.py]: Removed references of deployment config from variable names and comments

**Review by ramkiperiy — APPROVED** [review]

**Review by am-agrawa — APPROVED** [review]

**Pull request merged**